server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -168,10 +189,20 @@ public static ServerAddress startServer(MetricsSystem metricsSystem, ServerConte
           addresses);
     } catch (TTransportException e) {
       if (portSearch) {
+        // Build a list of reserved ports - as identified by properties of type PropertyType.PORT
+        Map<Integer,Property> reservedPorts = getReservedPorts(config);
+
         HostAndPort last = addresses[addresses.length - 1];
         // Attempt to allocate a port outside of the specified port property
         // Search sequentially over the next 1000 ports
         for (int port = last.getPort() + 1; port < last.getPort() + 1001; port++) {
+          if (reservedPorts.keySet().contains(port)) {
```

Comment:
May be able to do the following.

```suggestion
          if (reservedPorts.containsKey(port)) {
```

Reply:
Will do! Allow me a few hours to add a commit and push this change.

Reply:
I'm going to merge now. I can make this change subsequently.